localstack/services/awslambda/provider.py

```diff
@@ -467,16 +478,43 @@ def _validate_layers(new_layers: list[str]):
-            if layer is None:
-                raise InvalidParameterValueException(
-                    f"Layer version {layer_version_arn} does not exist.", Type="User"
-                )
-            layer_version = layer.layer_versions.get(layer_version)
-            if layer_version is None:
-                raise InvalidParameterValueException(
-                    f"Layer version {layer_version_arn} does not exist.", Type="User"
-                )
+            state = lambda_stores[layer_account_id][layer_region]
+            layer = state.layers.get(layer_name)
+            if layer_account_id == get_aws_account_id():
+                if region and layer_region != region:
+                    raise InvalidParameterValueException(
+                        f"Layers are not in the same region as the function. "
+                        f"Layers are expected to be in region {region}.",
+                        Type="User",
+                    )
+                if layer is None or layer.layer_versions.get(layer_version) is None:
+                    raise InvalidParameterValueException(
+                        f"Layer version {layer_version_arn} does not exist.", Type="User"
+                    )
+            else:  # External layer from other account
+                # TODO: validate IAM layer policy here, allowing access by default for now and only checking region
+                if region and layer_region != region:
+                    # TODO: detect user or role from context when IAM users are implemented
+                    user = "user/localstack-testing"
+                    raise AccessDeniedException(
+                        f"User: arn:aws:iam::{account_id}:{user} is not authorized to perform: lambda:GetLayerVersion on resource: {layer_version_arn} because no resource-based policy allows the lambda:GetLayerVersion action"
+                    )
+                if layer is None:
+                    # Limitation: cannot fetch external layers when using the same account id as the target layer
+                    # because we do not want to trigger the layer fetcher for every non-existing layer.
+                    if self.layer_fetcher is None:
+                        raise NotImplementedError(
+                            "Fetching shared layers from AWS is a pro feature."
+                        )
+
+                    layer = self.layer_fetcher.fetch_layer(layer_version_arn)
+                    if layer is None:
+                        # TODO: detect user or role from context when IAM users are implemented
+                        user = "user/localstack-testing"
+                        raise AccessDeniedException(
+                            f"User: arn:aws:iam::{account_id}:{user} is not authorized to perform: lambda:GetLayerVersion on resource: {layer_version_arn} because no resource-based policy allows the lambda:GetLayerVersion action"
+                        )
```

Comment:
I am still not 100% happy with this, as IAM should enforce it, but seems fine to leave it for now.

Reply:
Ok, maybe better to tackle this in a follow-up ...